stdlib/ballerina-builtin/src/main/java/org/ballerinalang/nativeimpl/file/Copy.java

```diff
@@ -45,85 +57,149 @@
         args = {@Argument(name = "source", type = TypeKind.STRUCT, structType = "File",
                 structPackage = "ballerina.file"),
                 @Argument(name = "destination", type = TypeKind.STRUCT, structType = "File",
-                        structPackage = "ballerina.file")},
+                          structPackage = "ballerina.file"),
+                @Argument(name = "replaceExisting", type = TypeKind.BOOLEAN)},
+        returnType = {
+                @ReturnType(type = TypeKind.STRUCT, structType = "FileNotFoundError", structPackage = "ballerina.file"),
+                @ReturnType(type = TypeKind.STRUCT, structType = "AccessDeniedError", structPackage = "ballerina.file"),
+                @ReturnType(type = TypeKind.STRUCT, structType = "IOError", structPackage = "ballerina.file")},
         isPublic = true
 )
 public class Copy extends AbstractNativeFunction {
 
-    private static final Logger logger = LoggerFactory.getLogger(Copy.class);
+    private static final Logger log = LoggerFactory.getLogger(Copy.class);
+    private static final PrintStream console = System.err;
     
     @Override
     public BValue[] execute(Context context) {
-
         BStruct source = (BStruct) getRefArgument(context, 0);
         BStruct destination = (BStruct) getRefArgument(context, 1);
+        boolean replaceExisting = getBooleanArgument(context, 0);
 
-        File sourceFile = new File(source.getStringField(0));
-        File destinationFile = new File(destination.getStringField(0));
+        Path sourceFile = Paths.get(source.getStringField(0));
+        Path destinationFile = Paths.get(destination.getStringField(0));
 
-        if (!sourceFile.exists()) {
-            throw new BallerinaException("failed to copy file: file not found: " + sourceFile.getPath());
-        }
-        File parent = destinationFile.getParentFile();
-        if (parent != null && !parent.exists() && !parent.mkdirs()) {
-            throw new BallerinaException("failed to copy file: cannot create directory: " + parent.getPath());
+        if (!Files.exists(sourceFile)) {
+            return getBValues(FileUtils.createFileNotFoundError(context, "Failed to copy file: '"
+                    + sourceFile.toString() + "'. File not found."), null, null);
         }
-        if (!copy(sourceFile, destinationFile)) {
-            throw new BallerinaException("failed to copy file: " + sourceFile.getPath());
+
+        try {
+            copyFile(sourceFile, destinationFile, replaceExisting);
+        } catch (SecurityException e) {
+            String errMsg = "Failed to copy file/dir: " + sourceFile.toString() + " to " + destinationFile.toString();
```

Comment:
Shall we use a string variable for - "Failed to copy file/dir: " as it is redundant?

Reply:
Ack.